src/notification.c

```diff
@@ -382,6 +363,122 @@ check_notification (GVariant *notification,
   return TRUE;
 }
 
+static gboolean
+validate_icon_more (GVariant *v)
+{
+  g_autoptr(GIcon) icon = g_icon_deserialize (v);
+  GBytes *bytes;
+  g_autofree char *name = NULL;
+  const char *argv[] = { LIBEXECDIR "/xdg-desktop-portal-validate-icon", "name", NULL };
+  xdp_autofd int fd = -1;
+  g_autoptr(GOutputStream) stream = NULL;
+  gssize written;
+  int status;
+  g_autofree char *err = NULL;
+  g_autoptr(GError) error = NULL;
+
+  if (G_IS_THEMED_ICON (icon))
+    {
+      g_autofree char *a = g_strjoinv (" ", (char **)g_themed_icon_get_names (G_THEMED_ICON (icon)));
+      g_debug ("Icon validation: themed icon (%s) is ok", a);
+      return TRUE;
+    }
+
+  if (!G_IS_BYTES_ICON (icon))
```

Comment:
Do we want to convert bytes icons into a local file? I mentioned it in the other comment that this would fix `xdg-desktop-portal-kde` which doesn't support these.

Reply:
does the kde backend support files ?

Reply:
Docs are kinda bad: https://api.kde.org/frameworks/knotifications/html/classKNotification.html#a9533d08af1499da4da7e1c92b8cd3b70

@grulja Does `KNotification::setIconName()` only support an icon theme name or will a file path work?

Reply:
It seems to work also with a file path. I just tested it and it works.

Reply:
Does it handled `file://` URIs? (which the freedesktop spec says to use)

Reply:
Sorry for not responding earlir, I somehow missed this question. Yes, it does handle file:// URIs.

Reply:
@grulja: the more important question is: do you handle GBytesIcon, with embedded png content?
 If not, could you add it?

Reply:
Unfortunately no, we don't handle it yet. KNotification API allows to set QPixmap so I guess in this case if QPIxmap is set, I should convert it to GBytesIcon and send over DBus. What is GBytesIcon, would something like this https://wiki.qt.io/Converting_QPixmap_to_QByteArray work? Just converting QPixmap to QByteArray and sending it over DBus?

Reply:
Yes that looks like it could work, but we don't guarantee that it is PNG so it might be JPEG or SVG, I assume Qt has APIs to figure that out. You can get the bytes via `g_bytes_icon_get_bytes()`.

Reply:
I've added docs about the supported formats now